### PR TITLE
Delete version items

### PR DESCRIPTION
**Pull request by hwilling**
fixes to issues #34 and #42, add ability for characters to unequip and remove items. allow deletion of classes, races, items, version info and stats in DB

### CODE REVIEW

File: src/main/kotlin/edu/ycp/cs482/iorcapi/factories/DetailFactory.kt [hunk 1]
```diff
@@ -102,6 +107,11 @@ class DetailFactory(
         return ClassQL(rpgClass)
     }
 
+    fun deleteClass(id:String):String{
+        classRepository.delete(id)
```
Comment (YoungMaker) [inline src/main/kotlin/edu/ycp/cs482/iorcapi/factories/DetailFactory.kt]: What does this do if the id doesn't exist? Probably nothing. Should look into that

File: src/main/kotlin/edu/ycp/cs482/iorcapi/factories/ItemFactory.kt [hunk 2]
```diff
@@ -20,6 +20,11 @@ class ItemFactory(
         return ItemQL(item)
     }
 
+    fun deleteItem(id:String):String{
+        itemRepository.delete(id)
```
Comment (YoungMaker) [inline src/main/kotlin/edu/ycp/cs482/iorcapi/factories/ItemFactory.kt]: See previous comment

File: src/main/kotlin/edu/ycp/cs482/iorcapi/resolvers/ItemMutationResolver.kt [hunk 3]
```diff
@@ -12,4 +12,9 @@ class ItemMutationResolver(
     fun addItemToVersion(name: String, description: String, price: Float,
     itemClasses: List<String>, version: String, type: ObjType)
         = itemFactory.addItem(name, description,price, itemClasses, version, type)
+
+    fun removeItem(id: String):String{
+        itemFactory.deleteItem(id)
```
Comment (YoungMaker) [inline src/main/kotlin/edu/ycp/cs482/iorcapi/resolvers/ItemMutationResolver.kt]: Another one. Need to deal with the consequences of trying to delete nonexistant items

File: src/main/kotlin/edu/ycp/cs482/iorcapi/factories/DetailFactory.kt (hunk 1 above)
Comment (hwilling) [reply src/main/kotlin/edu/ycp/cs482/iorcapi/factories/DetailFactory.kt]: Nothing on either the server or graphql outputs

File: src/main/kotlin/edu/ycp/cs482/iorcapi/factories/ItemFactory.kt (hunk 2 above)
Comment (hwilling) [reply src/main/kotlin/edu/ycp/cs482/iorcapi/factories/ItemFactory.kt]: same as previous

File: src/main/kotlin/edu/ycp/cs482/iorcapi/resolvers/ItemMutationResolver.kt (hunk 3 above)
Comment (hwilling) [reply src/main/kotlin/edu/ycp/cs482/iorcapi/resolvers/ItemMutationResolver.kt]: same as previous

**Pull request merged**